### ISSUE TITLE
[New build - DEFAULT] 22w02a: Bounty Hunting Missions

### ISSUE THREAD

**Issue by tsunamayo** [tsunamayo/Starship-EVO#4472]
Bounty Hunting missions have been added to the game.
- They spawn at the mission board at regular interval.
- For now if you want to see a difficulty rating you need to dock your ship at the station
- They will pay a cash reward.
- Upgrade Chips rewards are given to the ship the player is currently flying or last flown
- A new Weapon Score have been introduced. Fow now it is simply the power required by weapons divided by 10.
- More pirate ships have been added! Thanks to all the builders.
- Mission text is procedural.
- Foe name are also procedural.
- Mission are available for around 20min, but once you start it there is no time limit (yet)

Still Wip or to-do:
- Save mission state
- Assign target in a different system
- Assign several targets
- Give several locations to check

Hotfixes:
#4440 #4402 Warp issue when start far away from the star.
#4443 Event gate not triggering Indicators.
#4446 Event gate percentage input is confusing.
#4459 Round thruster incorrect mirroring.
#4419 Cannot build block after some mechanism copy-paste.

**Comment by tsunamayo** [comment]
Build is up! Thanks guys, there will be a few following update regarding this mission as I mostly had to focus on UI / reward stuff.
Also I will have to do some build request for what I call "Deep Space Point of Interest", as for now I only spawn a lovely Escape Pod courtesy of Smets.
Cheers

**Comment by pinesh** [comment]
Missions? Exiting! 

Will we be able to utilize these new spawning and generation mechanics for the custom lua quest work?

**Comment by tsunamayo** [comment]
@pinesh Well I still needs to do a bit of work on the framework but that is the plan yes...

**Comment by pinesh** [comment]
> @pinesh Well I still needs to do a bit of work on the framework but that is the plan yes...

Finally a reason for me to learn lua. Can’t wait to see more!

**Comment by ProPeach** [comment]
Here comes the first gameplay loop! Big moment for sevo, the year is off to a brilliant start 😄

**Comment by Uncle-Ulty** [comment]
Omg! there were TWO PIRATE ships (this is unfair! rofl!) ! Servo's and Surge's pirates! I had to retreat, repair and attack again (kiting) to defeat both!

I think I need a bigger ship for these missions...

![image](https://user-images.githubusercontent.com/73253472/149646510-c50fc329-c390-41b1-8f04-fa892f6dc311.png)

**Comment by tsunamayo** [comment]
> Here comes the first gameplay loop! Big moment for sevo, the year is off to a brilliant start 😄

Well it is not quite a complete loop just yet, but that's the first mission yes! I didnt worked at all on the mission itself though, more on the framework, but other one should be a lot faster to include...

**Comment by tsunamayo** [comment]
> Omg! there were TWO PIRATE ships (this is unfair! rofl!) ! Servo's and Surge's pirates! I had to retreat, repair and attack again (kiting) to defeat both!
> 
> I think I need a bigger ship for these missions...
> 
> ![image](https://user-images.githubusercontent.com/73253472/149646510-c50fc329-c390-41b1-8f04-fa892f6dc311.png)

Haha, congrats, you are the first to report the ambush. Speaking of which I forgot to lower the ambush percentage, it needs to be lower like 10%, for now it is 100% as I used that for testing... I will push a fix soon ^^

**Comment by Uncle-Ulty** [comment]
> > Omg! there were TWO PIRATE ships (this is unfair! rofl!) ! Servo's and Surge's pirates! I had to retreat, repair and attack again (kiting) to defeat both!
> > I think I need a bigger ship for these missions...
> > ![image](https://user-images.githubusercontent.com/73253472/149646510-c50fc329-c390-41b1-8f04-fa892f6dc311.png)
> 
> Haha, congrats, you are the first to report the ambush. Speaking of which I forgot to lower the ambush percentage, it needs to be lower like 10%, for now, it is 100% as I used that for testing... I will push a fix soon ^^

ROFL! Yeah, I survived cuz I made the kiting strategy: hit, run, hit, run! if these pirates have followed me, I'd be screwed!!!

This makes me think... maybe all the pirates should fully restore if you ran too far away...

**Comment by COMICLER** [comment]
Thanks, i just learned by ship is trash but im a great pilot
![20220115215627_1](https://user-images.githubusercontent.com/78708738/149646925-e78b278c-873d-4f85-b88c-296e69c915b7.jpg)

**Comment by Uncle-Ulty** [comment]
I'm using a 100t ships, and the enemies are melting my shields, rofl... gonna use a category 4 for this bounty...

**Comment by Uncle-Ulty** [comment]
The pirates were supposed to shoot each other? I think a friendly fire is causing them to fire each other.

![image](https://user-images.githubusercontent.com/73253472/149647755-bdb0bfb3-243c-4d8e-a4f9-99d51ab1f5df.png)

**Comment by Uncle-Ulty** [comment]
If you land and undock the same ship in 3 different hangars, the mission board will create 3 different inputs for the ship...

![image](https://user-images.githubusercontent.com/73253472/149647955-198e9749-7610-4e74-b763-a272a4d4d66a.png)

**Comment by tsunamayo** [comment]
@Uncle-Ulty Thanks, I see it make sense for the friendly fire, I will fix this. I will have a look at the other issues as well. This docked entity will be changed for survival, you will just see a list of the ships currently spawned that belongs to you.

**Comment by Kaiser-Indrasil** [comment]
@tsunamayo Very cool gameplay loop!
One thing I noticed is that both pirates spawned in the same location which might overwhelm the player in some situations.

Also, I would love to have a targetable bounty hunter mission objective for the lightcruise autopilot!

**Comment by tsunamayo** [comment]
> @tsunamayo Very cool gameplay loop! One thing I noticed is that both pirates spawned in the same location which might overwhelm the player in some situations.
> 
> Also, I would love to have a targetable bounty hunter mission objective for the lightcruise autopilot!

Yes I will completely nuke my current implementation of HUD target and interaction, and you would be able to lock on to them.

**Comment by tsunamayo** [comment]
> [bug] If you land and undock the same ship in 3 different hangars, the mission board will create 3 different inputs for the ship...
> 
> ![image](https://user-images.githubusercontent.com/73253472/149647955-198e9749-7610-4e74-b763-a272a4d4d66a.png)

Well I cant reproduce that actually, if you have a log file to show that would be great maybe something happened. Thanks

**Comment by tsunamayo** [comment]
@Uncle-Ulty nevermind I was just able to reproduce ^^, you just need to open the windows several time, it is not linked to multiple undock. Should be easy to fix.
